src/fetching/provider/testing.rs

```diff
@@ -54,7 +55,7 @@ impl<P> TestProvider<P> {
     pub async fn block(&self) {
         let mut unblock = self.unblock.write().await;
         if unblock.is_none() {
-            *unblock = Some(channel().0);
+            *unblock = Some(broadcast::channel(1000).0);
```

Comment:
Unbounded -> `1000` capacity, @jbearer does this matter?

Reply:
and @imabdulbasit

Reply:
This should be fine, it's only used for tests anyways